src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,186 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == size()) {
+            linkLast(value);
+        } else {
+            linkBefore(value, findNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> wanted = findNode(index);
+        return wanted.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> wanted = findNode(index);
+        T oldValue = wanted.item;
+        wanted.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlink(findNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> wanted = first; wanted != null; wanted = wanted.next) {
+                if (wanted.item == null) {
+                    unlink(wanted);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> wanted = first; wanted != null; wanted = wanted.next) {
+                if (object.equals(wanted.item)) {
+                    unlink(wanted);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder builder = new StringBuilder("[");
+        Node<T> current = first;
+        while (current != null) {
+            builder.append(current.item);
+            if (current.next != null) {
+                builder.append(", ");
+            } else {
+                builder.append("]");
+            }
+            current = current.next;
+        }
+        return builder.toString();
+    }
+
+    private void linkLast(T value) {
+        final Node<T> initLast = last;
+        final Node<T> newNode = new Node<>(initLast, value, null);
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. It looks like you're doing it correctly, but please ensure that this is consistent throughout your code.